sno/working_copy/mysql_adapter.py

```diff
@@ -0,0 +1,222 @@
+from sno import crs_util
+from sno.schema import Schema, ColumnSchema
+
+from sqlalchemy.dialects.mysql.base import MySQLIdentifierPreparer, MySQLDialect
+
+
+_PREPARER = MySQLIdentifierPreparer(MySQLDialect())
+
+
+V2_TYPE_TO_MYSQL_TYPE = {
+    "boolean": "bit(1)",
+    "blob": "longblob",
+    "date": "date",
+    "float": {0: "float", 32: "float", 64: "double precision"},
+    "geometry": "geometry",
+    "integer": {
+        0: "int",
+        8: "tinyint",
+        16: "smallint",
+        32: "int",
+        64: "bigint",
+    },
+    "interval": "text",
+    "numeric": "numeric",
+    "text": "longtext",
+    "time": "time",
+    "timestamp": "timestamp",
+}
+
+
+MYSQL_TYPE_TO_V2_TYPE = {
+    "bit": "boolean",
+    "tinyint": ("integer", 8),
+    "smallint": ("integer", 16),
+    "int": ("integer", 32),
+    "bigint": ("integer", 64),
+    "float": ("float", 32),
+    "double": ("float", 64),
+    "double precision": ("float", 64),
+    "binary": "blob",
+    "blob": "blob",
+    "char": "text",
+    "date": "date",
+    "datetime": "timestamp",
+    "decimal": "numeric",
+    "geometry": "geometry",
+    "numeric": "numeric",
+    "text": "text",
+    "time": "time",
+    "timestamp": "timestamp",
+    "varchar": "text",
+    "varbinary": "blob",
+}
+
+for prefix in ["tiny", "medium", "long"]:
+    MYSQL_TYPE_TO_V2_TYPE[f"{prefix}blob"] = "blob"
+    MYSQL_TYPE_TO_V2_TYPE[f"{prefix}text"] = "text"
```

Comment:
```suggestion
MYSQL_TYPE_TO_V2_TYPE = {
    "bit": "boolean",
    "tinyint": ("integer", 8),
    "smallint": ("integer", 16),
    "int": ("integer", 32),
    "bigint": ("integer", 64),
    "float": ("float", 32),
    "double": ("float", 64),
    "double precision": ("float", 64),
    "binary": "blob",
    "blob": "blob",
    "char": "text",
    "date": "date",
    "datetime": "timestamp",
    "decimal": "numeric",
    "geometry": "geometry",
    "longblob": "blob",
    "longtext": "text",
    "mediumblob": "blob",
    "mediumtext": "text",
    "numeric": "numeric",
    "text": "text",
    "time": "time",
    "timestamp": "timestamp",
    "tinyblob": "blob",
    "tinytext": "text",
    "varchar": "text",
    "varbinary": "blob",
}
```

Reply:
Didn't do - rearranged slightly so as not to redefine this constant, but I still didn't write these out in full. I still think it's easier to read when it's shorter / has less redundant information